src/t/d6.clj

```diff
@@ -0,0 +1,136 @@
+(ns t.d6
+  (:gen-class)
+  (:require
+    [clojure.string :as cljstr]
+    [clojure.set :as cljset]
+    [clojure.core.match :refer [match]]))
+
+(def sample
+"....#.....
+.........#
+..........
+..#.......
+.......#..
+..........
+.#..^.....
+........#.
+#.........
+......#...")
+
+(defn parse-input
+  [input]
+  (let [lines (cljstr/split-lines input)
+        y-max (dec (count lines))
+        x-max (dec (count (first lines)))]
+    {:y-max y-max
+     :x-max x-max
+     :map
+     (->> lines
+          (map-indexed
+            (fn [y l] (->>
+                        l
+                        (map-indexed
+                          (fn [x c]
+                            (cond
+                              (= c \#) {:O (set [[x y]])}
+                              (= c \^) {:G [[x y] :up]})))
+                        (apply merge-with into)
+                        )))
+          (apply merge-with into))}))
+
+(defn new-pos
+  [[gx gy] g-dir]
+  (match g-dir
+    :up [[gx (dec gy)] :right]
+    :right [[(inc gx) gy] :down]
+    :down [[gx (inc gy)] :left]
+    :left [[(dec gx) gy] :up]))
+
+(defn one-step-move
+  [info]
+  (let [obstacles (get-in info [:map :O])
+        [g-pos g-dir] (get-in info [:map :G])
+        [new-pos new-dir] (new-pos g-pos g-dir)
+        occupied? (contains? obstacles new-pos)]
+    (if occupied?
+      (recur (assoc-in info [:map :G 1] new-dir))
+      [(assoc-in info [:map :G 0] new-pos) new-pos])))
+
+(defn outside
+  [[x y] x-max y-max]
+  (or (< x 0)
+      (< x-max x)
+      (< y 0)
+      (< y-max y))
+  )
+
+(defn d6p1
+  [input]
+  (let [info-start (parse-input input) ]
+    (loop [info info-start
+           Gpos (set [(get-in info-start [:map :G 0])])]
+      (let [[n-info n-pos] (one-step-move info)]
+        (if (outside n-pos (:x-max info) (:y-max info))
+          (count Gpos)
+          (recur n-info (conj Gpos n-pos)))))
+    ))
+
+
+(defn one-step-move-p2
```

Comment:
This looks like the exact same logic as part 1's, except written a little bit more cleanly because now you get two arguments. Am I missing something?

Reply:
no, it is exactly like part 1
The big diff is that I got cold feet re-writing it for part one with a "I am nearly there, I should check it works before re-writing it". Once it worked, I moved to part 2 and only looked back because I had to >.>